__tests__/unit/send-email.service.spec.ts

```diff
@@ -0,0 +1,132 @@
+      const requestBody = (await request.json()) as Record<string, unknown>;
+
+      const authHeader = request.headers.get('Authorization');
+      if (authHeader !== `Bearer ${RESEND_API_KEY}`) {
+        return new HttpResponse(JSON.stringify({ error: { message: 'Invalid API key' } }), { status: 401 });
+      }
+
+      return HttpResponse.json({
+        id: 'mock-email-id',
+        from: requestBody.from,
+        to: requestBody.to,
+      });
+    }),
+    http.all('*', ({ request }) => {
+      console.error(`[MSW] Request not in whitelist: ${request.method} ${request.url}`);
+      return HttpResponse.error();
+    }),
+  );
+
+  beforeAll(() => server.listen());
+
+  beforeEach(() => {
+    errorSpy = vitest.fn();
+
+    sendEmailService = new ResendSendEmailService(
+      { error: errorSpy, info: vi.fn() },
+      makeConfigMock({ EMAIL_NAME, EMAIL_FROM, RESEND_API_KEY }),
+    );
+  });
+
+  afterEach(() => {
+    server.resetHandlers();
+    vitest.clearAllMocks();
+  });
+
+  afterAll(() => server.close());
+
+  describe('sendEmail', () => {
+    it('successfully sends an email with HTML content', async () => {
+      const emailOptions: SendEmailOptions = {
+        to: ['recipient@example.com'],
+        title: 'Test Email',
+        html: '<p>This is a test email</p>',
+      };
+
+      await expect(sendEmailService.sendEmail(emailOptions)).resolves.not.toThrow();
+      expect(errorSpy).not.toHaveBeenCalled();
+    });
+
+    it('successfully sends an email with text content', async () => {
+      const emailOptions: SendEmailOptions = {
+        to: ['recipient@example.com'],
+        title: 'Test Email',
+        text: 'This is a test email',
+      };
+
+      await expect(sendEmailService.sendEmail(emailOptions)).resolves.not.toThrow();
+      expect(errorSpy).not.toHaveBeenCalled();
+    });
+
+    it('successfully sends an email to multiple recipients', async () => {
+      const emailOptions: SendEmailOptions = {
+        to: ['recipient1@example.com', 'recipient2@example.com'],
+        title: 'Test Email',
+        html: '<p>This is a test email</p>',
+      };
+
+      await expect(sendEmailService.sendEmail(emailOptions)).resolves.not.toThrow();
+      expect(errorSpy).not.toHaveBeenCalled();
+    });
+
+    it('throws an exception when the Resend API returns an error', async () => {
+      server.use(
+        http.post(RESEND_API_URL, async () => {
+          return new HttpResponse(JSON.stringify({ error: { message: 'Failed to send email' } }), { status: 400 });
+        }),
+      );
+
+      const emailOptions: SendEmailOptions = {
+        to: ['error@example.com'],
+        title: 'Test Email',
+        html: '<p>This is a test email</p>',
+      };
+
+      await expect(sendEmailService.sendEmail(emailOptions)).rejects.toThrow(Exception);
+      expect(errorSpy).toHaveBeenCalledWith({
+        err: {
+          error: {
+            message: 'Failed to send email',
+          },
+        },
+      });
```

Comment:
if you don't reuse it, you can pass the object directly.

```suggestion
      const emailOptions: SendEmailOptions = ;

      await expect(
        sendEmailService.sendEmail({
          to: ['error@example.com'],
          title: 'Test Email',
          html: '<p>This is a test email</p>',
        })
      ).rejects.toThrow(Exception);
      
      expect(errorSpy).toHaveBeenCalledWith({
        err: { error: { message: 'Failed to send email' } },
      });
```

Reply:
Done https://github.com/GenesisEducationKyiv/software-engineering-school-5-0-allohamora/pull/5/commits/526ef4ca61ac8b075631da8a6cf5cd94181b9a3f